core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -63,7 +63,6 @@ object RequestChannel extends Logging {
     // o.a.k.common.requests.AbstractRequest.getRequest()
     private val keyToNameAndDeserializerMap: Map[Short, (ByteBuffer) => RequestOrResponse]=
       Map(ApiKeys.FETCH.id -> FetchRequest.readFrom,
-        ApiKeys.METADATA.id -> TopicMetadataRequest.readFrom,
```

Comment:
Hmm, we removed similar methods from `Broker` in the rack-aware patch (Jun asked them to be removed) so I think it's OK to remove them from here too. It's not only internal API, but it's in the core JAR. If you'd rather keep it for now, that's OK too, it's just a pain to maintain if we rev the protocol.